cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,892 @@
+) -> np.ndarray:
+    """This method returns one score per labeled box in an image. Score from 0 to 1 ranking how likeley swapped the box is."""
+
+    auxiliary_input_dict = _get_valid_inputs_for_compute_scores_per_image(
+        alpha=alpha,
+        label=label,
+        prediction=prediction,
+        pred_labels=pred_labels,
+        pred_label_probs=pred_label_probs,
+        pred_bboxes=pred_bboxes,
+        lab_labels=lab_labels,
+        lab_bboxes=lab_bboxes,
+        similarity_matrix=similarity_matrix,
+        min_possible_similarity=min_possible_similarity,
+    )
+
+    pred_labels = auxiliary_input_dict["pred_labels"]
+    pred_label_probs = auxiliary_input_dict["pred_label_probs"]
+    lab_labels = auxiliary_input_dict["lab_labels"]
+    similarity_matrix = auxiliary_input_dict["similarity_matrix"]
+    min_possible_similarity = auxiliary_input_dict["min_possible_similarity"]
+
+    scores_swap = np.empty(
+        shape=[
+            len(lab_labels),
+        ]
+    )  # same length as number of labeled boxes
+    for iid, k in enumerate(lab_labels):
+        not_k_idx = pred_labels != k
+
+        if len(not_k_idx) == 0:
+            scores_swap[iid] = 1.0
+            continue
+
+        not_k_similarity = similarity_matrix[iid, not_k_idx]
+        not_k_pred = pred_label_probs[not_k_idx]
+
+        idx_at_least_high_probability_threshold = not_k_pred > high_probability_threshold
+        if len(idx_at_least_high_probability_threshold) == 0:
+            scores_swap[iid] = 1.0
+            continue
+
+        not_k_similarity = not_k_similarity[idx_at_least_high_probability_threshold]
+        if len(not_k_similarity) == 0:  # if there is no annotated box
+            scores_swap[iid] = 1.0
+        else:
+            closest_predicted_box = np.argmax(not_k_similarity)
+            score = np.max([min_possible_similarity, 1 - not_k_similarity[closest_predicted_box]])
+            scores_swap[iid] = score
+    return scores_swap
+
+
+def _compute_swap_box_scores(
+    alpha: float,
+    high_probability_threshold: float,
+    labels: Optional[List[Dict[str, Any]]] = None,
+    predictions: Optional[List[np.ndarray]] = None,
+    auxiliary_inputs: Optional[List[AuxiliaryTypesDict]] = None,
+) -> List[np.ndarray]:
+    if auxiliary_inputs is None:
+        auxiliary_inputs = _get_valid_inputs_for_compute_scores(alpha, labels, predictions)
+
+    scores_swap = []
+    for auxiliary_inputs in auxiliary_inputs:
+        scores_swap_per_box = _compute_swap_box_scores_for_image(
+            alpha=alpha, high_probability_threshold=high_probability_threshold, **auxiliary_inputs
+        )
+        scores_swap.append(scores_swap_per_box)
+    return scores_swap
+
+
+def _get_subtype_label_quality_scores(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    *,
+    alpha: float,
+    low_probability_threshold: float,
+    high_probability_threshold: float,
+    temperature: float,
```

Comment:
Should these kwargs have default values?

Reply:
@ulya-tkch Consider that we want to eventually make this method user-facing. In that case, we should have the default for alpha, low_probability_threshold, high_probability_threshold, temperature all be None

and in the code have eg:
```
if alpha = None:
    alpha = IMPORTED_ALPHA_CONSTANT
```